pkg/network/usm/istio.go

```diff
@@ -118,30 +182,31 @@ func newIstioMonitor(c *config.Config, mgr *manager.Manager) (*istioMonitor, err
 	return m, nil
 }
 
-// Start the istioMonitor
-func (m *istioMonitor) Start() {
-	if m == nil {
-		return
-	}
+// ConfigureOptions changes map attributes to the given options.
+func (m *istioMonitor) ConfigureOptions(options *manager.Options) {
+	sharedLibrariesConfigureOptions(options, m.cfg)
+}
 
+// PreStart is called before the start of the provided eBPF manager.
+func (m *istioMonitor) PreStart() error {
 	if m.attacher == nil {
-		log.Error("istio monitoring is enabled but the attacher is nil")
-		return
+		return errors.New("istio monitoring is enabled but the attacher is nil")
 	}
 
 	if err := m.attacher.Start(); err != nil {
-		log.Errorf("Cannot start istio attacher: %s", err)
+		return fmt.Errorf("cannot start istio attacher: %w", err)
 	}
 
-	log.Info("istio monitoring enabled")
+	return nil
+}
+
+// PostStart is called after the start of the provided eBPF manager.
+func (*istioMonitor) PostStart() error {
+	return nil
 }
 
 // Stop the istioMonitor.
 func (m *istioMonitor) Stop() {
-	if m == nil {
-		return
-	}
```

Comment:
Have you removed this in purpose? there is no need to validate that the monitor is not nil?

Reply:
yes, if there's a nil monitor then it is considered not-enabled/not-supported, and we will never reach to the interface methods